test/integration/hca_azul_client_test.rb

```diff
@@ -6,25 +6,47 @@ class HcaAzulClientTest < ActiveSupport::TestCase
 
   before(:all) do
     @hca_azul_client = ApplicationController.hca_azul_client
-    @project_shortname = 'HumanTissueTcellActivation'
-    @project_id = '4a95101c-9ffc-4f30-a809-f04518a23803'
+    @project_shortname = 'ImmuneCellExhaustianHIV'
+    @project_id = '0fd8f918-62d6-4b8b-ac35-4c53dd601f71'
     @facets = [
-      { id: 'disease', filters: [{ id: 'MONDO_0005109', name: 'HIV infectious disease' }] },
-      { id: 'species', filters: [{ id: 'NCBITaxon_9606', name: 'Homo sapiens' }] }
+      {
+        id: 'disease',
+        filters: [{ id: 'MONDO_0005109', name: 'HIV infectious disease' }],
+        db_facet: SearchFacet.find_or_create_by(identifier: 'disease', data_type: 'string')
+      },
+      {
+        id: 'species',
+        filters: [{ id: 'NCBITaxon_9606', name: 'Homo sapiens' }],
+        db_facet: SearchFacet.find_or_create_by(identifier: 'species', data_type: 'string')
+      }
     ]
     catalogs = @hca_azul_client.catalogs
     @default_catalog = catalogs['default_catalog']
-    SearchFacet.update_all_facet_filters
-  end
+    @query_json = {
+      sampleDisease: { is: ['HIV infectious disease'] },
+      genusSpecies: { is: ['Homo sapiens'] }
+    }.with_indifferent_access
 
-  after(:all) do
-    SearchFacet.all.map(&:update_filter_values!)
+    # check to see if Azul is up and responding as we expect it to
+    # the status information should indicate all services are up, and running a query for the ImmuneCellExhaustianHIV
+    # project should return exactly one match
+    # if either of these do not return correctly, or error, set a state to skip all tests w/o throwing an error
+    @azul_is_ok = false
+    begin
+      api_up = @hca_azul_client.api_available?
+      project = @hca_azul_client.projects(query: @query_json, size: 1)
+      if api_up && get_entries_from_response(project, :projects).present?
+        @azul_is_ok = true
+      end
+    rescue RestClient::Exception => e
+      puts "Error in determining if Azul is healthy: #{e.message}"
+    end
```

Comment:
I suspect we don't currently do it, but how feasible would it be to log such exceptions from the `test` environment to Mixpanel and/or Sentry?

My vague understanding is that Azul becomes not-fully-available for a while a day or two per week.  That's concerning for a service we'll (entirely?) depend on for global study search.

Metrics on the frequency of such degradations would inform decisions around reliability engineering allocation, etc.

Reply:
Feasible: very.  But I don't think Mixpanel would be the appropriate place for this.  Since this is specifically regarding the `test` environment, and is an actual error, Sentry would be the obvious place to log this.  We would also have the added benefit of having parameters/stack traces as well, in addition to more easily accessing times/counts of any outages.  Also, I don't think merging non-production analytics into our existing Mixpanel dashboards is a habit we want to get into.  Would Sentry tracking suffice for this?

Also, regarding Azul stability: it is unclear to me how much of a problem this actually is.  They only release catalogs on a monthly basis, and do normal deployments on a similar schedule to us.  That doesn't account for a lot of the problems we've had with builds lately.  In addition, when looking at recent build failures, only a portion of them were related to Azul.  Of those that were, it seems that we also had a possible issue with idempotency in some of the tests (i.e. they sometimes fail depending on the order in which they're run), which this PR tries to address.  My hope is that this PR will address those lurking issues, and will then allow us a good, clean signal on how much of an issue service interruptions from Azul are.

Reply:
Ah, I see where you're going with this now.  I agree - that makes sense, and seems like a good discussion topic.